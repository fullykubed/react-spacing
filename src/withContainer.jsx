import React from 'react';
import {maybeNull} from "./maybeNull";

export function withContainer(Component){
  return (props) => {
    const {className, style, ifNull, ...rest} = props;
    const SafeComponent = maybeNull({
      Component,
      ifNull,
      ifRender: (children) => (
        <div className={className} style={style}>
          {children}
        </div>
      )
    });
    return <SafeComponent {...rest}/>;
  }
}