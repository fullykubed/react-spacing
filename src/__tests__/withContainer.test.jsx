import React from 'react';
import {mount} from 'enzyme';

import {withContainer} from "../withContainer";

const TestComponent = withContainer(({requiredProp}) => {

  if(requiredProp === undefined) return null;

  return <p>I received {`${requiredProp}`}!</p>
});

function NullRender(){
  return <p>TestComponent was null!</p>
}

describe('withContainer', () => {

  it('doesnt add div on null render', () => {
    expect(mount(<TestComponent className={"blah"}/>)).toMatchSnapshot();
  });

  it('adds div', () => {
    expect(mount(<TestComponent className={"blah"} requiredProp={"Jack"}/>))
      .toMatchSnapshot();
  });

  it('executes alternative on null render', () => {
    expect(mount(<TestComponent ifNull={NullRender}/>)).toMatchSnapshot();
  });
});