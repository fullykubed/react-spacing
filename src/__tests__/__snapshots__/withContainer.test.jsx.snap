// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withContainer adds div 1`] = `
<bound 
  className="blah"
  requiredProp="Jack"
>
  <bound 
    requiredProp="Jack"
  >
    <div
      className="blah"
    >
      <p>
        I received 
        Jack
        !
      </p>
    </div>
  </bound >
</bound >
`;

exports[`withContainer doesnt add div on null render 1`] = `
<bound 
  className="blah"
>
  <bound  />
</bound >
`;

exports[`withContainer executes alternative on null render 1`] = `
<bound 
  ifNull={[Function]}
>
  <bound >
    <p>
      TestComponent was null!
    </p>
  </bound >
</bound >
`;
